### PR TITLE
Made config class compatible with new concat module

### LINKED ISSUE #109: puppetlabs-concat v1.2.1 with file_concat breaks sshd_config ~> Service
When using this module with the latest version of puppetlabs-concat the File/Service relationship fails.

```
File[$ssh::params::sshd_config] ~> Service[$ssh::params::service_name]
```

as the File resource doesn't exist. The new concat module doesn't create a File resource by the looks of things anymore. With some very loose testing I've found that 

1. Removing the File ~> Service relationship chains seems fine, as the Class ~> Class relationships between the 'ssh::server::config' and 'ssh::server::service' are defined.
2. Changing the File ~> Service relationship to Concat ~> Service works fine as the relationship is now with the public module name rather than the internal File resource.

I'm not completely sure which method would be the best.

### PR DESCRIPTION
**Pull request by tedivm**
This method has the benefit of being backwards compatible as well as future compatible.

This should resolve #109.

### CODE REVIEW

**Comment by geoffgarside** [comment]
Is that subscribe the wrong way round? Wouldn't that try to refresh the concat when the service changes? Should it be `notify` instead?

Also I mentioned in #109 that theres a second `File ~> Service` relationship in the `manifests/server.pp` file as well as the `manifests/server/config.pp`

**Comment by tedivm** [comment]
That second relationship seems like a bug (or at least just redundant)- it's literally defining the same service relationships.

File: manifests/server/config.pp [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
   concat { $ssh::params::sshd_config:
-    ensure => present,
-    owner  => '0',
-    group  => '0',
-    mode   => '0600',
+    ensure    => present,
+    owner     => '0',
+    group     => '0',
+    mode      => '0600',
+    notify => Service[$ssh::params::service_name]
```
Comment (johanfleury) [inline manifests/server/config.pp]: `=>` should be aligned to avoid warning

**Comment by nibalizer** [comment]
I'm able to successfully use this.

**Comment by nibalizer** [comment]
The errors in travis are just lint errors, super simple to fix.

**Comment by tedivm** [comment]
I'll run it through puppet-lint when I'm home and update this.

**Comment by tedivm** [comment]
The issues puppet lint is finding are not a result of my code changes. As such I don't think it's appropriate for me to make changes.

**Pull request merged**